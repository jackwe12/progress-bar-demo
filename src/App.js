import React, {useState, useEffect} from 'react';
import Bar from './components/ProgressBar/ProgressBar';
import BarSelect from './components/BarSelect/BarSelect';
import ControlBtn from './components/ControlBtn/ControlBtn';
import './App.scss';

function App() {
  const [bar, setBar] = useState([]);
  const [buttons, setButtons] = useState([]);
  const [limit, setLimit] = useState(0);
  const [current, setCurrent] = useState(0); 

  useEffect(()=>{
    fetch(`https://pb-api.herokuapp.com/bars`)
    .then(response=>response.json())    
    .then(data=>{                
      setBar(data.bars);
      setButtons(data.buttons);
      setLimit(data.limit);
      })
  }, [])

  const handleScore = (index, delta) => {
    
    let changedValue = bar[index] + delta;

    if (changedValue > 0 || delta > 0)   
    {
      setBar([...bar.slice(0, index), bar[index] + delta, ...bar.slice(index+1)])
    }

    else if (changedValue < 0 )
    {
      setBar([...bar.slice(0, index), 0, ...bar.slice(index+1)])
    }


  }   

  return (
    <>
      <h1 className='title center'>Progress Bars Demo</h1>
      <h2 className='limit center'>Limit: {limit}</h2>
      <div className="bar-section">
        {bar.map((barValue, index)=>{
        return(
          <Bar
            barValue = {barValue}
            limit = {limit}
            key = {index}
          />
        )
        })}

      </div>
      <div className='control-section'>
        <div className='control-section-inside'>
            <div className='bar-select-button '>
            <BarSelect
              current = {current}
              setCurrent = {setCurrent}
              bars = {bar}
            /> 
          </div>
          <div className='controlBtn-section '>
            {buttons.map((buttonValue, index)=>{
            return(
              <ControlBtn
                buttonValue = {buttonValue}
                index = {index}
                handleScore = {handleScore}
                current = {current}
                key={index}
              />
            )

            })}
          </div>
        </div>   
      </div>


    </>
  );
}

export default App;
