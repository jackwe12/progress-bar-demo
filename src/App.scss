.center{
    margin:auto;
    width:50%
}
.title{
    font-size: 3rem;
    margin-top: 3rem;
}
.limit{
    font-size: 2.5rem;
    margin-top: 1rem;
    margin-bottom: 1rem;

}
.progress{
    margin: auto;
    margin-bottom: 0.5rem;
}
.control-section{

    &-inside{
        margin:auto;
        width:50%;
        display: flex;
        flex-wrap: wrap;
    }
}

@media screen and (max-width:500px){
.title{
    font-size: 2rem;
    width: 80%;
}
.limit{
    font-size:1.5rem;
    width: 80%;
}

.control-section{
    .bar-select-button{
        margin:auto;
        margin-bottom: 1rem;
    }
    .controlBtn-section{
        margin:auto;
    }
    &-inside{
        width:100%;
    }
}

}
@media screen and (max-width:350px){
.title{
    font-size:1.7rem;
}
.limit{
    font-size:1.3rem;
    width: 80%;
}

}