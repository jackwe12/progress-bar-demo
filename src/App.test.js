import App from './App';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });



describe('Testing <App/>', ()=>{
    test('snapshot testing',()=>{
        const tree =renderer
            .create(     <App />)
            .toJSON();
            expect(tree).toMatchSnapshot();
    })
});
