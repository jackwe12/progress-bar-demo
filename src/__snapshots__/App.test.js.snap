// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App/> snapshot testing 1`] = `
Array [
  <h1
    className="title center"
  >
    Progress Bars Demo
  </h1>,
  <h2
    className="limit center"
  >
    Limit: 
    0
  </h2>,
  <div
    className="bar-section"
  />,
  <div
    className="control-section"
  >
    <div
      className="control-section-inside"
    >
      <div
        className="bar-select-button "
      >
        <span>
          <div
            className="dropdown"
            onKeyDown={[Function]}
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-success"
              disabled={false}
              id="dropdown-basic"
              onClick={[Function]}
              type="button"
            >
              # Progress
              1
            </button>
          </div>
        </span>
      </div>
      <div
        className="controlBtn-section "
      />
    </div>
  </div>,
]
`;
